fast64_internal/sm64/sm64_geolayout_classes.py

```diff
@@ -467,7 +468,7 @@ def walk(node, last_materials):
                 cmd_list = fMesh.drawMatOverrides.get(base_node.override_hash, None) or fMesh.draw
                 lastMat = last_materials.get(base_node.drawLayer, None)
                 default_render_mode = fModel.getRenderMode(base_node.drawLayer)
-                lastMat = self.bleed_fmesh(fModel.f3d, fMesh, lastMat, cmd_list, default_render_mode)
+                lastMat = self.bleed_fmesh(fModel.f3d, fMesh, lastMat if not base_node.bleed_independently else None, cmd_list, default_render_mode)
```

Comment:
it's hard to follow which type ``base_node`` can be there but I'm concerned about it being ``None``, in this case I'd say I'm not ok with using a bool-like condition when the value is None but if you're sure it's never that value and it's true/false then it's fine

Reply:
`base_node` is never `None`.
The construction of geo layout nodes starts with a `TransformNode` class, which holds an attribute `node` always. If `node` is none then the geo layout node is invalid because it has no data.
As for what `base_node` is, it is a subclass of `BaseDisplayListNode` because those are the classes that have a FMesh defined.

Reply:
I was talking about ``bleed_independently`` which gets initialised as None (my bad I should have precised that earlier)

Reply:
This should be valid code if `bleed_independently` is `None` as `None` resolves to False in any test of truthyness.
I can make it init as `False` without really changing things if that makes more sense.

Reply:
I commented on that because I remembered reading something Dragorn said in codebase todo (https://discord.com/channels/874816709855440926/887690239911526431/888901683730595891)

I think that would be the case here

Reply:
I do believe it is always explicitly set but I'll change it to `False` anyway since that is more inline with the expected value.